packages/assets-controllers/src/TokensController.ts

```diff
@@ -52,54 +48,23 @@ export interface TokensConfig extends BaseConfig {
   provider: any;
 }
 
-/**
- * @type AssetSuggestionResult
- * @property result - Promise resolving to a new suggested asset address
- * @property suggestedAssetMeta - Meta information about this new suggested asset
- */
-interface AssetSuggestionResult {
-  result: Promise<string>;
-  suggestedAssetMeta: SuggestedAssetMeta;
-}
-
-enum SuggestedAssetStatus {
-  accepted = 'accepted',
-  failed = 'failed',
-  pending = 'pending',
-  rejected = 'rejected',
-}
-
-export type SuggestedAssetMetaBase = {
-  id: string;
-  time: number;
-  type: string;
-  asset: Token;
-  interactingAddress?: string;
-};
-
 /**
  * @type SuggestedAssetMeta
  *
  * Suggested asset by EIP747 meta data
- * @property error - Synthesized error information for failed asset suggestions
  * @property id - Generated UUID associated with this suggested asset
- * @property status - String status of this this suggested asset
  * @property time - Timestamp associated with this this suggested asset
  * @property type - Type type this suggested asset
  * @property asset - Asset suggested object
  * @property interactingAddress - Account address that requested watch asset
  */
-export type SuggestedAssetMeta =
-  | (SuggestedAssetMetaBase & {
-      status: SuggestedAssetStatus.failed;
-      error: Error;
-    })
-  | (SuggestedAssetMetaBase & {
-      status:
-        | SuggestedAssetStatus.accepted
-        | SuggestedAssetStatus.rejected
-        | SuggestedAssetStatus.pending;
-    });
+export type SuggestedAssetMeta = {
+  id: string;
+  time: number;
+  type: string;
+  asset: Token;
+  interactingAddress: string;
```

Comment:
The only place where this object is created is in the method watchAsset and it uses:
```
interactingAddress: interactingAddress || selectedAddress
```
The value of selectedAddress is defined as a string, not string | undefined, so this value is never undefined.